src/read.c

```diff
@@ -2809,8 +2840,23 @@ static avifResult avifParseItemPropertyAssociation(avifMeta * meta, const uint8_
                 avifProperty * dstProp = (avifProperty *)avifArrayPush(&item->properties);
                 AVIF_CHECKERR(dstProp != NULL, AVIF_RESULT_OUT_OF_MEMORY);
                 *dstProp = *srcProp;
+
+                if (isTransformative) {
+                    transformativePropertySeen = AVIF_TRUE;
+                }
             } else {
+                AVIF_ASSERT_OR_RETURN(!isTransformative);
+
                 if (essential) {
+                    // ISO/IEC 23008-12 Section 10.2.1:
+                    //   Under any brand, the primary item (or an alternative if alternative support is required)
+                    //   shall be processable by a reader implementing only the required features of that brand.
+                    //   Specifically, given that each brand has a set of properties that a reader is required to
+                    //   support: the item shall not have properties that are marked as essential and are outside
+                    //   this set.
+                    // Assuming this rule also applies to items whose primary item depends on (such as the cells
```

Comment:
Nit: Assuming => Assume ?

Nit: The word "whose" sounds strange here. Should it be changed to "the"?